books/precalculus/bake

```diff
@@ -0,0 +1,168 @@
+      BakeNumberedExercise.v1(exercise: exercise, number: exercise.count_in(:chapter))
+    end
+    chapter.pages.search('section.section-exercises').exercises.each do |exercise|
+      BakeNumberedExercise.v1(exercise: exercise, number: exercise.count_in(:page))
+    end
+
+    # # Bake answer key from chapter/ move solutions from eoc into answer key
+    MoveSolutionsToAnswerKey.v1(chapter: chapter, metadata_source: metadata,
+                                strategy: :precalculus, append_to: solutions_container,
+                                solutions_plural: false)
+  end
+
+  book.pages('$:not(.appendix)').each do |page|
+    page.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v2(table: table,
+                           number: table.count_in(:page))
+    end
+    page.figures.each do |figure|
+      BakeFigure.v1(figure: figure,
+                    number: figure.count_in(:page))
+    end
+    page.examples.each do |example|
+      BakeExample.v1(example: example,
+                     number: example.count_in(:page),
+                     title_tag: 'h3')
+    end
+  end
+
+  book.composite_pages.each do |page|
+    page.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v2(table: table,
+                           number: table.count_in(:composite_page))
+    end
+    page.figures('$:not(.unnumbered)').each do |figure|
+      BakeFigure.v1(figure: figure,
+                    number: figure.count_in(:composite_page))
+    end
+  end
+
+  book.pages('$.appendix').each do |page|
+    appendix_letter = [*('A'..'Z')][page.count_in(:book) - 1]
+    BakeAppendix.v1(page: page, number: appendix_letter)
+
+    page.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v2(table: table, number: "#{appendix_letter}#{table.count_in(:page)}")
+    end
+    page.figures('$:not(.unnumbered)').each do |figure|
+      BakeFigure.v1(figure: figure, number: "#{appendix_letter}#{figure.count_in(:page)}")
+    end
+  end
+
+  BakeUnnumberedTables.v1(book: book)
+
+  # Add classes to exercises and solutions - TODO: think of a better way to do this
+  book.search('section.review-exercises, section.practice-test, section.section-exercises,' \
+    ' div[data-type="note"].try').exercises.each do |exercise|
+    BakeFirstElements.v1(within: exercise)
+    inline_list = exercise.solution&.first('div.os-solution-container > ol[type="1"]:first-child')
+    if inline_list
+      inline_list.add_class('first-inline-list-element')
+      inline_list.parent.add_class('has-first-inline-list-element')
+    end
+  end
+
+  book.search('div.os-solution-area > div[data-type="solution"]').each do |solution|
+    BakeFirstElements.v1(within: solution)
+    inline_list = solution&.first('div.os-solution-container > ol[type="1"]:first-child')
+    if inline_list
+      inline_list.add_class('first-inline-list-element')
+      inline_list.parent.add_class('has-first-inline-list-element')
+    end
+  end
```

Comment:
there could be room for optimization here -- one way would be to make these calls part of `BakeNumberedExercises` to avoid querying the classes twice. 
- These classes do have to be added after tables are baked, because they look for the `os-table` class.
- It does seem like the `within` targeting has to be pretty specific though